packages/lib/services/e2ee/EncryptionService.ts

```diff
@@ -106,14 +109,47 @@ export default class EncryptionService {
 		return this.loadedMasterKeyIds().length;
 	}
 
-	public chunkSize() {
-		return this.chunkSize_;
+	// Note for methods using SJCL:
+	//
+	// 1 MB is very slow with Node and probably even worse on mobile.
+	//
+	// On mobile the time it takes to decrypt increases exponentially for some reason, so it's important
+	// to have a relatively small size so as not to freeze the app. For example, on Android 7.1 simulator
+	// with 4.1 GB RAM, it takes this much to decrypt a block;
+	//
+	// 50KB => 1000 ms
+	// 25KB => 250ms
+	// 10KB => 200ms
+	// 5KB => 10ms
+	//
+	// So making the block 10 times smaller make it 100 times faster! So for now using 5KB. This can be
+	// changed easily since the chunk size is incorporated into the encrypted data.
+	public chunkSize(method: EncryptionMethod) {
+		type EncryptionMethodChunkSizeMap = Record<EncryptionMethod, number>;
+		const encryptionMethodChunkSizeMap: EncryptionMethodChunkSizeMap = {
+			[EncryptionMethod.SJCL]: 5000,
+			[EncryptionMethod.SJCL1a]: 5000,
+			[EncryptionMethod.SJCL1b]: 5000,
+			[EncryptionMethod.SJCL2]: 5000,
+			[EncryptionMethod.SJCL3]: 5000,
+			[EncryptionMethod.SJCL4]: 5000,
+			[EncryptionMethod.Custom]: 5000,
+			[EncryptionMethod.KeyV1]: 5000, // Master key is not encrypted by chunks so this value will not be used.
+			[EncryptionMethod.FileV1]: 131072, // 128k
```

Comment:
(comment because of #9852, my `joplin-desktop` now contains 4G data + 8G encrypted data, feel free to ignore!)

for such large chunk size, do you think the encrypted bytes might benefit from a quick zstd compression? i.e. tentatively compress after `encryptRaw` then set `result.compressed = true; result.ct = encryptedCompressed` if smaller, the same for decrypt

Reply:
I'm not sure if the extra compression step will slow down the encryption/decryption speed. The E2EE on mobile platform is not very fast, so the speed should be taken into consideration.
With that said, even if we don't use compression now, the encrypted file size is decreased in the new encryption methods. ([report](https://discourse.joplinapp.org/t/final-report-of-the-native-encryption-project/40171#p-128759-h-4-reduce-the-size-overhead-of-encryption-5)).

Reply:
zstd can reach around 500mb/s with default compression level on desktop, while I can't find a benchmark on mobile, its overhead(a slight regression over huge improvements you brought) should remain manageable 

I'm still reading the full report, overhead decrease in file size and master key are promising, thank you!

Reply:
The speed overhead could be much greater than expected.
For example, the raw AES encryption speed on desktop is over a few GB/s
https://discourse.joplinapp.org/t/performance-test-of-node-js/38762#aes-encryptiondecryption-performance-2
But in the final implementation, the encryption speed is less than 100MB/s
https://discourse.joplinapp.org/t/final-report-of-the-native-encryption-project/40171#p-128759-based-on-commit-edd2a9002a31b0fee4f11ded0a37951543e289ab-release-build-7

Things could be worse on React Native. The raw AES encryption speed could reach 100MB/s. But in the final implementation, the encryption speed is less than 2MB/s
https://discourse.joplinapp.org/t/performance-test-of-react-native-quick-crypto/38622#aes-encryptiondecryption-performance-2

Reply:
I see, I almost never edit notes on my phone, but the decryption slowness on first sync was already ~~concerningly~~ impressive..